src/shogun/kernel/CombinedKernel.h

```diff
@@ -124,7 +124,14 @@ class CCombinedKernel : public CKernel
 		 */
 		inline CKernel* get_kernel(int32_t idx)
 		{
-			return (CKernel*) kernel_array->get_element(idx);
+			if (idx < get_num_kernels())
+			{
+				return (CKernel*)kernel_array->get_element(idx);
+			}
+			else
+			{
+				return 0;
```

Comment:
I think this is bad, an error should be thrown if the index is bad (either from the array, or by this class). 
Also if you do bound checks, also assert for >=0